File: app/src/main/java/jp/sourceforge/gokigen/memoma/preference/PreferenceIntentCaller.java
package jp.sourceforge.gokigen.memoma.preference;

import android.app.Activity;
import android.content.Intent;

public class PreferenceIntentCaller implements IPreferenceIntentCaller
{
    private final Activity activity;

    public static IPreferenceIntentCaller newInstance(Activity activity)
    {
        return (new PreferenceIntentCaller(activity));
    }

    private PreferenceIntentCaller(Activity activity)
    {
        this.activity = activity;
    }


    @Override
    public boolean selectBackgroundImageFileFromGallery(int code)
    {
        Intent intent = new Intent(Intent.ACTION_PICK);
        intent.setType("image/*");
        activity.startActivityForResult(intent, code);
        return (false);
    }
}

File: app/src/main/java/jp/sourceforge/gokigen/memoma/preference/PreferenceIntentCaller.kt
package jp.sourceforge.gokigen.memoma.preference

import android.content.Intent
import androidx.fragment.app.FragmentActivity

class PreferenceIntentCaller private constructor(private val activity: FragmentActivity?) :
    IPreferenceIntentCaller {
    override fun selectBackgroundImageFileFromGallery(code: Int): Boolean {
        val intent = Intent(Intent.ACTION_PICK)
        intent.setType("image/*")
        activity?.startActivityForResult(intent, code)
        return (false)
    }

    companion object {
        fun newInstance(activity: FragmentActivity?): IPreferenceIntentCaller {
            return (PreferenceIntentCaller(activity))
        }
    }
}